src/oscillators/sa5x_oscillator.c

```diff
@@ -21,20 +21,60 @@
 #define ATTR_FW_SERIAL  1
 #define ATTR_CTRL       2
 #define ATTR_STATUS     4
+#define ATTR_PHASE		8
 
 #define CMD_SWVER                     "\\{swrev?}"
 #define CMD_SERIAL                    "{serial?}"
 #define CMD_GET_LOCKED                "{get,Locked}"
+#define CMD_GET_GNSS_PPS              "{get,PpsInDetected}"
 #define CMD_GET_PHASE                 "{get,Phase}"
 #define CMD_GET_LASTCORRECTION        "{get,LastCorrection}"
 #define CMD_GET_TEMPERATURE           "{get,Temperature}"
 #define CMD_GET_ANALOG_TUNING         "{get,AnalogTuning}"
 #define CMD_GET_DIGITAL_TUNING        "{get,DigitalTuning}"
 #define CMD_GET_ANALOG_TUNING_ENABLED "{get,AnalogTuningEnabled}"
+#define CMD_GET_TAU                   "{get,TauPps0}"
+#define CMD_SET_TAU                   "{set,TauPps0,%d}"
+#define DISCIPLINING_PHASES 3
+
+static const unsigned int tau_values[DISCIPLINING_PHASES] = {50, 500, 10000};
+static const unsigned int tau_interval[DISCIPLINING_PHASES] = {600, 7200, 86400}; // in seconds
+
+enum SA5x_ClockClass {
+	SA5X_CLOCK_CLASS_UNCALIBRATED,
+	SA5X_CLOCK_CLASS_CALIBRATING,
+	SA5X_CLOCK_CLASS_HOLDOVER,
+	SA5X_CLOCK_CLASS_LOCK,
+	SA5X_CLOCK_CLASS_NUM
+};
+
+enum SA5x_Disciplining_State {
+	/** Initialization State */
+	SA5X_INIT,
```

Comment:
Do you really expect TRACKING, LOCK_LOW_RESOLUTION and LOCK_HIGH_RESOLUTION states for SA5x disciplining, or is it just to align with what is done for the mRO-50 ?

Reply:
It was done just to align with disciplinning-minipod values because these values are used in monitoring output.
I don't like the idea of using constants/enums from different libraries in pluggable modules (like oscillators) that's why I replicated the statuses. Better solution could be to export strings from oscillator modules itself and move all the dependencies to oscillators, but it will require a lot of refactoring

Reply:
I though the monitoring just needed abstract states "disciplining" and "lock", and not the inner disciplining states. Is it not the case ?

Reply:
We are trying to "mimic" to what is made for mro50 so that's why I had to implement the same statuses.
@CharlesParent @HugoFolcher @mattlrtz Could you please merge this PR? and make a new version tag for us to build new packages